vm/src/stdlib/itertools.rs

```diff
@@ -195,8 +195,8 @@ mod decl {
             vm: &VirtualMachine,
         ) -> PyResult {
             let start: PyObjectRef = start.into_option().unwrap_or_else(|| vm.new_pyobj(0));
-            let step: PyIntRef = step.into_option().unwrap_or_else(|| vm.new_pyref(1));
-            if !PyNumber::check(&start, vm) {
+            let step: PyObjectRef = step.into_option().unwrap_or_else(|| vm.new_pyobj(1));
```

Comment:
```suggestion
            let step = step.into_option().unwrap_or_else(|| vm.new_pyobj(1));
```
The type will be inferred by `step`